<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GMainWindow</class>
 <widget class="QMainWindow" name="GMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLineEdit" name="fontLineEdit">
        <property name="text">
         <string notr="true">/usr/share/fonts/noto/NotoSerif-Regular.ttf</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="fontSelButton">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="fontSizeBox">
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="minimum">
         <double>6.000000000000000</double>
        </property>
        <property name="maximum">
         <double>72.000000000000000</double>
        </property>
        <property name="value">
         <double>12.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Antialiasing:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="antialiasBox">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <item>
         <property name="text">
          <string>None</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Gray</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Hinting:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="hintingBox">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <item>
         <property name="text">
          <string>No</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Bytecode</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Auto</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Gamma:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="gammaBox">
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>5.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="kerningBox">
        <property name="text">
         <string>Use kerning</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="ligaturesBox">
        <property name="text">
         <string>Allow ligatures</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLineEdit" name="sampleTextLineEdit">
      <property name="text">
       <string notr="true">The quick brown fox jumps over the lazy dog.</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="GLowLevelTextRender" name="previewWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>100</horstretch>
        <verstretch>100</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <addaction name="menu_File"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_Exit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLowLevelTextRender</class>
   <extends>QWidget</extends>
   <header>lowlevel_textrender.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>action_Exit</sender>
   <signal>triggered()</signal>
   <receiver>GMainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sampleTextLineEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_textEdited(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontSizeBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_fontSizeChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontLineEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_fontFileEdited(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontSelButton</sender>
   <signal>clicked()</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_browseFont()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>antialiasBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_antialiasingChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gammaBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_gammaChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>367</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hintingBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_hintingChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>kerningBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_kerningChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>452</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ligaturesBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>GMainWindow</receiver>
   <slot>slot_ligaturesChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>539</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slot_textEdited(QString)</slot>
  <slot>slot_fontSizeChanged(double)</slot>
  <slot>slot_fontFileEdited(QString)</slot>
  <slot>slot_browseFont()</slot>
  <slot>slot_antialiasingChanged(int)</slot>
  <slot>slot_gammaChanged(double)</slot>
  <slot>slot_hintingChanged(int)</slot>
  <slot>slot_kerningChanged(int)</slot>
  <slot>slot_ligaturesChanged(int)</slot>
 </slots>
</ui>
